### PR TITLE
Homework2(Terekh Vasyl) Var4

### PR DESCRIPTION
**Pull request by Vasyl78**
…eak' and class method 'default_car'. Created classes 'Desktop', 'Laptop' and module 'LinuxFriendly'. Prepended module in class 'Laptop'

### CODE REVIEW

File: lib/topics/state_and_behaviour.rb [hunk 1]
```diff
@@ -2,4 +2,31 @@
...
 module StateAndBehaviour
+  # class Car / Default car "Chargo"
+  class Car
+    attr_accessor :color, :year, :model, :current_speed
+
+    def initialize(params)
+      @color = params[:color] || 'red'
+      @model = params[:model] || 'Chargo'
+      @year = params[:year] || 1981
+      @current_speed = 0
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb]: 0 - magic_number, use CONSTANTS (and same for values before)

File: lib/topics/state_and_behaviour.rb [hunk 2]
```diff
@@ -2,4 +2,31 @@
...
+    end
+
+    def self.default_car
+      Car.new
+    end
+
+    def speed_up(speed)
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb]: more_speed will better describe what up should do

File: lib/topics/state_and_behaviour.rb [hunk 3]
```diff
@@ -2,4 +2,31 @@
...
+      @current_speed += speed
+    end
+
+    def push_break(speed)
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb]: try less_speed

File: lib/topics/state_and_behaviour.rb [hunk 4]
```diff
@@ -2,4 +2,31 @@
...
+      if speed <= @current_speed
```
Comment (sveredyuk) [inline lib/topics/state_and_behaviour.rb]: what about to do this in one line?

Comment (Vasyl78) [reply lib/topics/state_and_behaviour.rb]: So right?

(speed <= @current_speed) ? (@current_speed -= less_speed) : @current_speed

File: lib/topics/state_and_behaviour.rb (hunk 1 above)
Comment (Vasyl78) [reply lib/topics/state_and_behaviour.rb]: when I use constants rubocop an error "dynamic constant assignment"
      START_SPEED = 0
      DEFAULT_COLOR = "red"
      DEFAULT_MODEL = "Chargo"
      DEFAULT_YEAR = 1981
Why? And how can I fix error?

File: lib/topics/state_and_behaviour.rb (hunk 4 above)
Comment (sveredyuk) [reply lib/topics/state_and_behaviour.rb]: yes, but use readers for current_spead instead @ivars (less dogs - less issues)

File: lib/topics/state_and_behaviour.rb (hunk 1 above)
Comment (Vasyl78) [reply lib/topics/state_and_behaviour.rb]: I fixed all errors

**Pull request closed without merging**